pkgs/build-support/php/hooks/composer-install-hook.sh

```diff
@@ -0,0 +1,72 @@
+declare -a composerVendorCache
+declare -a installComposerDevDependencies
+declare -a runComposerScripts
+declare -a runComposerPlugins
+declare -a dontComposerInstallConfigureHook
+declare -a dontComposerInstallBuildHook
+declare -a dontComposerInstallInstallHook
+declare -a version
+
+if [[ -z "${dontComposerInstallConfigureHook-}" ]]; then
+  preConfigureHooks+=(composerInstallConfigureHook)
+fi
+
+if [[ -z "${dontComposerInstallBuildHook-}" ]]; then
+  preBuildHooks+=(composerInstallBuildHook)
+fi
+
+if [[ -z "${dontComposerInstallInstallHook-}" ]]; then
+  preInstallHooks+=(composerInstallInstallHook)
+fi
+
+composerInstallConfigureHook() {
+    echo "Executing composerInstallConfigureHook"
+
+    cp -ar ${composerVendorCache}/repo .
+    cp -ar ${composerVendorCache}/composer.lock .
+    composer config repo.packagist false
+    composer config repo.local '{"type": "composer", "url": "file:///build/source/repo"}'
```

Comment:
Probably safer to use `$PWD/repo`. The name of the source directory will depend on what is passed to `src`.